[GeoMechanicsApplication] Marked a few member functions `[[noreturn]]`

avdg81 opened the pull request:
**📝 Description**
This PR solves three new issues detected by SonarQube.

avdg81 committed:
Marked a few member functions `[[noreturn]]`

This solves three new issues detected by SonarQube.

rfaasse reviewed (dismissed):
Makes sense, thanks for the quick PR 👍

WPK4FEM reviewed (dismissed):
Looks correct to me.

avdg81 committed:
Attempt to fix build errors using Clang or GCC

Moved the `[[noreturn]` attribute to the first declaration of the member functions.

WPK4FEM reviewed (approved):
Still looks O.k. for me.
